test/streambuf_readstream.cpp

```diff
@@ -7,3 +7,40 @@
 
 // Test that header file is self-contained.
 #include <beast/streambuf_readstream.hpp>
+
+#include <beast/streambuf.hpp>
+#include <beast/detail/unit_test/suite.hpp>
+#include <boost/asio.hpp>
+
+namespace beast {
+
+class streambuf_readstream_test : public beast::detail::unit_test::suite
+{
+public:
+    void testSpecial()
+    {
+        using socket_type = boost::asio::ip::tcp::socket;
+        boost::asio::io_service ios;
+        {
+            streambuf_readstream<socket_type, streambuf> srs(ios);
+            streambuf_readstream<socket_type, streambuf> srs2(std::move(srs));
+            srs = std::move(srs2);
+        }
+        {
+            socket_type sock(ios);
+            streambuf_readstream<socket_type&, streambuf> srs(sock);
+            streambuf_readstream<socket_type&, streambuf> srs2(std::move(srs));
```

Comment:
Ah. I was pattern matching with the previous block, and missed that this declared with a reference.